lhotse/supervision.py

```diff
@@ -64,6 +66,34 @@ def filter(self, predicate: Callable[[SupervisionSegment], bool]) -> 'Supervisio
         """
         return SupervisionSet.from_segments(seg for seg in self if predicate(seg))
 
+    def find(
+            self,
+            recording_id: str,
+            channel: int,
+            start_after: Seconds = 0,
+            end_before: Optional[Seconds] = None,
+            adjust_offset: bool = False
+    ) -> Iterable[SupervisionSegment]:
+        """
+        Return an iterable of segments that match the provided ``recording_id``.
+
+        :param recording_id: Desired recording ID.
+        :param channel: Return supervisions in the specified channel.
+        :param start_after: When specified, return segments that start after the given value.
+        :param end_before: When specified, return segments that end before the given value.
+        :param adjust_offset: When true, return segments as if the recordings had started at ``start_after``.
+            In other words, :math:`segment.start = segment.start - start_after`.
+        :return: An iterator over supervision segments satisfying all criteria.
+        """
+        return (
+            segment.with_offset(-start_after) if adjust_offset else segment
+            for segment in self
+            if segment.recording_id == recording_id
```

Comment:
wouldn't this be extremely slow for big manifests?  And if start_after and end_before meant to be filters can you clarify?  start_after seems to have a side effect of modifying the start time.  And I'm not sure the logic is correct if start_after and end_before are both nonzero.

Reply:
Yes, that could get quite slow. There was another place in the code with the same issue - I believe it's in `FeatureSet.find()`. I'd leave it be for now until it hurts us, and once it does, we can optimize it. I'm thinking about constructing and caching an index: something like a `Dict[<recording-id>, IntervalTree[Segment]]`, where the interval tree uses (start, end) to make the search in the time axis efficient.

Reply:
Re other comments - I updated the description, but the idea is that `start_after` *may* modify the start time, but only if you explicitly request it with `adjust_offset=True`. This is to make it easier to create Cuts. I want the supervisions in the Cuts to have start/end times relative to the start of the Cut, and not to the start of the underlying Recording. I expect it to be more useful that way.

Reply:
I added a bunch of unit tests to check the logic - I think it's fine?

Reply:
My feeling is just a dict on the recording-id, initially, would be enough.